### ISSUE TITLE
Collection expression conversion failure should result in element-wise diagnostics

### ISSUE THREAD

**Issue by RikkiGibson** [dotnet/roslyn#73297]
SharpLab is out of date so you'll have to paste into a console app or similar.

```cs
A[] arr = [
    new A(),
    new B() // CS0029: Cannot implicitly convert type 'B' to 'A' (squiggling only 'new B()')
];

M([
    new A(),
    new B()
]); // CS1503: Argument 1: cannot convert from 'collection expressions' to 'A[]' (squiggling the entire collection-expr)

void M(A[] arr) { }

class A { }
class B { }
```

**Expected behavior**: both cases report errors on the specific elements which are failing to convert.

**Actual behavior**: Argument to `M` gives an error on the entire collection-expr. One can imagine this being *unwieldy* for *large collection-exprs*. Only assignment to `A[] arr` reports element-wise errors in the sample.

It's possible that element-wise conversion errors are not always what we want. But, I am suggesting that we have consistency between conversion diagnostics for assignment and for argument conversions, as a baseline.

I think the reason for the current behavior with arguments has to do with the way overload resolution errors are reported. But it feels like even in overload cases, we should be able to report fine-grained diagnostics, when the candidate we choose for diagnostics doesn't have a conversion from collection-expr, and the parameter type is a collection type.
